docs/design/design-rtsp-record-application.md

```diff
@@ -0,0 +1,130 @@
+# RTSP recording application 
+
+User can use this application to watch rtsp using ffmpeg or other third-party package to store the stream into PV in other host which has RTX 4090.
+
+## Goal
+
+application running in the kubernetes allow user to connect RTSP server, and record the rtsp stream into file like mp4 / flv and so on, user can export file out of kubernetes.
+
+## General Design
+
+The application is a HTTP Server running in the kubernetes, which can store the rtsp stream into a file like mp4 / flv and so on, user can export file out of kubernetes with the instruction manual.
+
+
+
+## Detailed Design
+
+### Management RTSP Servers
+
+application manage a map to record the info of each device which record and can save this map into a file in PV for Backups, When The Application is up, it will recover from backups first.
+
+user can register his device using `/register`, so that application need using a Kubernetes to get this namespace's secret.
+
+rbac.yaml
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: application
+  namespace: shifu-app
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: [ "get", "list", "watch", "update", "create","delete"]
+```
+
+### Store the Stream
+
+We need to store the video in pieces, each clip is 1 hour by default, the user can set this option, user can customize the path where the files saved
+
+this application could using the third-party package like ffmpeg to implement it or using `exec.Command`
+ffmpeg command example:
+```bash
+ffmpeg -i rtsp://[RTSP_URL] -c copy -map 0 -segment_time 300 -f segment video/output%03d.mp4
+```
+
+Create PVC and PV need Separate with other deploy to avoid user delete pv by mistake. 
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: <devicename>-pv
+  namespace: deviceshifu
+  labels:
+    pv: <devicename>-pv
+spec:
+  capacity:
+    storage: 10Gi
+  volumeMode: Filesystem
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: slow
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: <devicename>-pvc
+spec:
+  resources:
+    requests:
+      storage: 10Gi
+  volumeMode: Filesystem
+  accessModes:
+    - ReadWriteOnce
+  selector:
+    matchLabels:
+      pv: <devicename>-pv
+```
+recording server deployment.yaml
+```yaml
+# deployment.yaml
+...
+- env:
+  RTSPServerAddress: ""
+  UserName: ""
+  PasswordSecret: "" # this just a case, secret using the secret env mode
+  FileSavePath: ""
```

Comment:
do we still need this if we have a config file

Reply:
no need, i will remove it